source/server/overload_manager_impl.h

```diff
@@ -109,6 +111,11 @@ class OverloadManagerImpl : Logger::Loggable<Logger::Id::main>, public OverloadM
   absl::node_hash_map<std::string, Resource> resources_;
   absl::node_hash_map<std::string, OverloadAction> actions_;
 
+  absl::flat_hash_map<std::string, OverloadActionState> state_updates_to_flush_;
+  absl::flat_hash_map<ActionCallback*, OverloadActionState> callbacks_to_flush_;
+  int flush_epoch_ = 0;
+  int flush_awaiting_updates_ = 0;
```

Comment:
nit: prefer unsigned here and elsewhere. I would use uint64_t or size_t if you prefer.